### PR TITLE
Sort out documentation for both API versions

### PR DESCRIPTION
**Pull request by lyzadanger**
Fixes https://github.com/hypothesis/product-backlog/issues/976

This PR spins off a new OpenAPI document for v2, expands the version documentation in each and starts to reorganize the sphinx-generated documentation so we have docs for both versions of the API (with strong warnings about v2).

The main changes here are to the "Using the Hypothesis API" docs page in Sphinx:

image

And the addition of a second API reference, for v2:

image

Other than version numbers and some intro text, the v1 and v2 API description documents (and hence the rendered docs) are identical.

### CODE REVIEW

**Comment by lyzadanger** [comment]
(removed WIP; ready for review)

**Comment by lyzadanger** [comment]
I'm marking this WIP because I don't want to merge it until after #5590, but it is ready for review.

File: docs/_extra/api-reference/hypothesis-v1.yaml [hunk 1]
```diff
@@ -56,6 +44,34 @@ info:
     X-Hypothesis-Media-Type: application/vnd.hypothesis.v1+json
```
Comment (robertknight) [inline docs/_extra/api-reference/hypothesis-v1.yaml]: As you noted on the call, the `X-` prefix needs to be removed to match the implementation.

File: docs/_extra/api-reference/hypothesis-v1.yaml [hunk 2]
```diff
@@ -14,29 +14,17 @@ info:
...
-    See [the API overview](../api/) for an introduction to the API and information
-    on how authorization works.
-
-    ## Expanding Resources
-
-    Resources often contain links to related resources in their response bodies.
-    Some endpoints support resource expansion, allowing those related resources
-    to be expanded inline within the response.
 
+    See [the API overview](https://h.readthedocs.io/en/latest/api/)
```
Comment (robertknight) [inline docs/_extra/api-reference/hypothesis-v1.yaml]: Was there a particular reason for changing to an absolute URI here? I presume this is copied from the `hypothesis-v2.yaml` docs?

File: docs/_extra/api-reference/index.html [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 <html>
   <head>
-    <title>Hypothesis API documentation</title>
+    <title>Hypothesis API documentation (v1)</title>
     <meta name="viewport" content="width=device-width, initial-scale=1">
   </head>
   <body>
-    <redoc spec-url="hypothesis.yaml"></redoc>
-    <script src="https://cdn.jsdelivr.net/npm/redoc@2.0.0-alpha.17/bundles/redoc.standalone.js"></script>
+    <redoc spec-url="hypothesis-v1.yaml" expand-responses="200,201"></redoc>
```
Comment (robertknight) [inline docs/_extra/api-reference/index.html]: `hypothesis.yaml` is still in the source tree. Should it be removed? I found one remaining reference to it in `docs/developing/documentation.rst`.

File: docs/_extra/api-reference/hypothesis-v1.yaml (hunk 2 above)
Comment (lyzadanger) [reply docs/_extra/api-reference/hypothesis-v1.yaml]: Yes! This OpenAPI document is standalone—it should work outside of the context of our docs site.

**Pull request merged**